plugins/candela/web_client/views/CandelaWidget.js

```diff
@@ -44,61 +45,77 @@ var CandelaWidget = View.extend({
     },
 
     render: function () {
-        if (this.accessLevel >= AccessType.READ && this.item.get('name').toLowerCase().endsWith('.csv')) {
-            this.$el.html(CandelaWidgetTemplate({
-                components: this._components
-            }));
-            this.parametersView.setElement($('.g-item-candela-parameters'));
-            datalib.csv(this.item.downloadUrl(), (error, data) => {
-                if (error) {
-                    let info = {
-                        text: 'An error occurred while attempting to read and ' +
-                              'parse the data file. Details have been logged in the console.',
-                        type: 'danger',
-                        timeout: 5000,
-                        icon: 'attention'
-                    };
-                    events.trigger('g:alert', info);
-                    console.error(error);
-                    return;
-                }
+        let cleanup = function () {
+            this.$('.g-item-candela')
+                .remove();
+        };
+
+        if (this.accessLevel < AccessType.READ) {
+            cleanup();
+            return;
+        }
+
+        let parser = null;
+        let name = this.item.get('name').toLowerCase();
+        if (name.endsWith('.csv')) {
+            parser = datalib.csv;
+        } else if (name.endsWith('.tsv') || name.endsWith('.tab')) {
+            parser = datalib.tsv;
+        } else {
+            cleanup();
+            return;
+        }
+
+        this.$el.html(CandelaWidgetTemplate({
+            components: this._components
+        }));
+        this.parametersView.setElement($('.g-item-candela-parameters'));
+        parser(this.item.downloadUrl(), (error, data) => {
```

Comment:
If candela has a promise-style calling mode, we should use that instead. If not, this is OK for now.